### PR TITLE
Use extra context for runtime VM

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
We now partially revert https://github.com/cri-o/cri-o/pull/4650 to
still use the runtime internal context for its tasks.

#### Which issue(s) this PR fixes:

Fixes: https://github.com/cri-o/cri-o/issues/4798
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Fixed bug where runtime VM created containers never reach their `completed` state. 
```

### CODE REVIEW

**Comment by fidencio** [comment]
@saschagrunert, thanks for the PR, it does solve the issue I reported.

tl;dr: LGTM, we should proceed with this PR, and I'd like to hear from @haircommander before adding my `lgtm`.

Now, the long reply ...

It bogus my mind, sincerely, and I've gone through this on Friday, coming up with pretty much the same solution proposed by in this PR, then I filed the issue.  I still don't fully get why we need the extra context, why the context coming from the GRPC doesn't do what it has to do.

@saschagrunert mentioned this may be a bug in the GRPC implementation of the runtime VM, and he may very much be right about that.  For now, I'm looking forward to have this partial revert merged, and backported to the `release-1.21` branch, as it'll unblock the 2.1.0-rc0 release of kata-containers.  Then we can track down what's broken on the runtime VM side. :-/

**Comment by haircommander** [comment]
thanks for tracking this down y'all
